### PR TITLE
update substitutions during subinplace

### LINKED ISSUE #585: apply constraint.subinplace to substitutions
As it stands, substitutions may get stranded during a varkey/varkey substitution.

### PR DESCRIPTION
**Pull request by bqpd**
requires #858, closes #585

### CODE REVIEW

**Comment by bqpd** [comment]
still doesn't work on `aircraft.py`...

**Comment by bqpd** [comment]
@whoburg, ready for review

File: gpkit/constraints/set.py [hunk 1]
```diff
@@ -198,11 +200,26 @@ def subinplace(self, subs):
             if subs_:
                 subs_["filter"] = "ConstraintSet"
                 constraint.subinplace(subs_)
+                allsubs_.update(subs)
```
Comment (whoburg) [inline gpkit/constraints/set.py]: inline explaining what `allsubs_` is?

File: gpkit/constraints/set.py [hunk 2]
```diff
@@ -198,11 +200,26 @@ def subinplace(self, subs):
...
+            if k in self.substitutions:
+                if isinstance(v, Iterable) and k.shape:
+                    keys = self.varkeys.keymap[k]
+                    v = np.array(v) if not hasattr(v, "shape") else v
+                    for ki in keys:
+                        self.substitutions[v[ki.idx]] = self.substitutions[ki]
+                else:
+                    try:
+                        self.substitutions[v.key] = self.substitutions[k]
+                    except:
```
Comment (whoburg) [inline gpkit/constraints/set.py]: I would not have expected pylint to allow this -- specify exception type?

**Comment by whoburg** [comment]
retest this please

**Comment by galbramc** [comment]
test this please

**Pull request merged**